uhabits-android/src/main/java/org/isoron/uhabits/activities/habits/list/views/CheckmarkButtonView.kt

```diff
@@ -80,15 +82,24 @@ class CheckmarkButtonView(
         setOnLongClickListener(this)
     }
 
-    fun performToggle() {
+    private fun performToggle() {
         value = Entry.nextToggleValue(
             value = value,
             isSkipEnabled = preferences.isSkipEnabled,
             areQuestionMarksEnabled = preferences.areQuestionMarksEnabled
         )
-        onToggle(value)
-        performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
-        invalidate()
+
+        if (preferences.isSkipEnabled) {
+            {
+                onToggle(value)
+                performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
+                invalidate()
+            }.delay(TOGGLE_DELAY_MILLIS)
+        } else {
+            onToggle(value)
+            performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
+            invalidate()
+        }
```

Comment:
1. We can apply the delay in all cases (not only when skips are enabled).
2. I just realized that this solution unfortunately breaks the ripple effect. The ripple effect should play immediately after the checkmark is entered, but it's currently playing seconds after. I think a better place to add the delay might be around line 143 of HabitCardView.kt.